### PR TITLE
create network with a custom id when force param is set

### PR DESCRIPTION
**Pull request by Lupino**
In my situation I use htm.core rest_server to predict and anomly the temperature which from real temperature sensor.

The sensor on my iot platform has an unique id, which can be an network id.

If i has 1000 sensor,  on the old way i create a table which contain unique id and network id.
when the rest_server restart the network id is reset,  then the unique id and network id map is failure.

### CODE REVIEW

**Comment by Lupino** [comment]
> What you propose is a valid way to do this. And thanks for contributing.
> 
> But I wonder if we can improve on this a little.
> Consider this set of situations for Network object creation:
> 
> 1. No id is provided  ( POST /network )
>    In this case the we assign the next available id, create a new instance of the Network object and put it in the table, and return the id.
> 2. An id is provided but it does not already exist in the table.
>    In this case we create using this id and insert a new Network object into the table.
> 3. An id is provided and it does already exists
>    In this case we re-use the id.  Create a new instance of the Network table and replace the exiting one.
> 
> By doing this we avoid the extra 'force' argument.
> Unless I miss-understand, this accomplishes what you were trying to achieve.
> If it is, then we can adjust the logic accordingly.

Think for advice

**Comment by breznak** [comment]
@Lupino something failed in tests in Debug mode, but I was unable to view the CI log, so rerunning all the tests. Please keep an eye on it.

**Comment by Lupino** [comment]
```
cmake -DCMAKE_BUILD_TYPE=Debug ..
```

is this enter Debug Mode?

```
[----------] Global test environment tear-down
[==========] 370 tests from 42 test cases ran. (101650 ms total)
[  PASSED  ] 370 tests.

  YOU HAVE 2 DISABLED TESTS
```

I run on macos

**Comment by breznak** [comment]
> is this enter Debug Mode? 

yes, that's correct. 

> [==========] 370 tests from 42 test cases ran. (101650 ms total)
[  PASSED  ] 370 tests. 

everything passing OK in CI now too. Some hiccup in the CI/logs, sorry.

File: src/examples/rest/server_core.hpp [hunk 1]
```diff
@@ -89,12 +89,13 @@ class RESTserver {
 
     //  POST /network?id=<previous id>
     //  Configure a Network resource (with JSON configuration in POST body) ==> token
-    //  Note: the id parameter is optional.  If given it will re-use (replace) a previous id.
+    //  Note: the id parameter is optional.  If given it will re-use (replace) a previous id or use it.
```
Comment (breznak) [inline src/examples/rest/server_core.hpp:92]: nit, maybe clarify "(replace) a previous id or create a new one with the id" ?

**Comment by dkeeney** [comment]
We do need a unit test to confirm that it would use the id if provided rather than create a new id.  However, I think I would like to create a new PR that changes the calling arguments slightly and I can include a new test in that PR.

So lets merge this change.   Thank you @Lupino for your contribution.

And I will make a new PR for another change.  What I would like to change is to allow the Network object creation to use either the syntax:
```
   POST /network?id=<previous_id>
```
or the syntax:
```
   POST /network/<previous_id>
```
or
```
   POST /network
```
This last one will just use the next available id.  The other two will use the given id.
The objective will be to avoid confusion as to which syntax is supported by allowing all three.

**Comment by breznak** [comment]
> This last one will just use the next available id. The other two will use the given id.
The objective will be to avoid confusion as to which syntax is supported by allowing all three. 

good solution :+1: the disambiguity was my only "dislike" on this PR.

**Pull request merged**